File: C#/Trivia/Trivia/Questions.cs
using System;
using System.Collections.Generic;
using System.Linq;

namespace Trivia
{
    public class Questions
    {
        LinkedList<string> popQuestions = new LinkedList<string>();
        LinkedList<string> scienceQuestions = new LinkedList<string>();
        LinkedList<string> sportsQuestions = new LinkedList<string>();
        LinkedList<string> rockQuestions = new LinkedList<string>();

        public Questions()
        {
            GenerateQuestions();
        }

        private void GenerateQuestions()
        {
            for (var i = 0; i < 50; i++)
            {
                popQuestions.AddLast("Pop Question " + i);
                scienceQuestions.AddLast(("Science Question " + i));
                sportsQuestions.AddLast(("Sports Question " + i));
                rockQuestions.AddLast("Rock Question " + i);
            }
        }

        public void AskQuestion(int location)
        {
            var currentCategory = CurrentCategory(location);
            Console.WriteLine($"The category is {currentCategory}");
            if (currentCategory == "Pop")
            {
                Console.WriteLine(popQuestions.First());
                popQuestions.RemoveFirst();
            }
            if (currentCategory == "Science")
            {
                Console.WriteLine(scienceQuestions.First());
                scienceQuestions.RemoveFirst();
            }
            if (currentCategory == "Sports")
            {
                Console.WriteLine(sportsQuestions.First());
                sportsQuestions.RemoveFirst();
            }
            if (currentCategory == "Rock")
            {
                Console.WriteLine(rockQuestions.First());
                rockQuestions.RemoveFirst();
            }
        }

        private String CurrentCategory(int location)
        {
            if (location % 4 == 0) return "Pop";
            if (location % 4 == 1) return "Science";
            if (location % 4 == 2) return "Sports";
            return "Rock";
        }
    }
}

File: php/src/Domain/Questions.php
<?php

namespace Trivia\Domain;

class Questions
{
    private $questions;

    /**
     * Questions constructor.
     */
    public function __construct()
    {
        $this->questions[0] = array();
        $this->questions[1] = array();
        $this->questions[2] = array();
        $this->questions[3] = array();

        for ($i = 0; $i < 50; $i++) {
            array_push($this->questions[0], new Question("Pop", "Pop Question " . $i));
            array_push($this->questions[1], new Question("Science", "Science Question " . $i));
            array_push($this->questions[2], new Question("Sports", "Sports Question " . $i));
            array_push($this->questions[3], new Question("Rock", "Rock Question " . $i));
        }
    }

    public function drawQuestion($location): Question
    {
        return array_shift($this->questions[$location % 4]);
    }
}